lib/models/ui/challenge_card_data.dart

```diff
@@ -0,0 +1,41 @@
+import "package:flutter/foundation.dart";
+
+@immutable
+class ChallengeCardData {
+  final String title;
+  final int? distance;
+  final int? timeLeft;
+  final int reward;
+
+  /// Creates a [ChallengeCardData] with the given parameters. The [title] is the
+  /// post's title, the [distance] is the distance to the challenge in meters, the [timeLeft]
+  /// is the time left to complete the challenge in hours, and the [reward] is the reward
+  /// for completing the challenge.
+  /// If the [distance] is `null`, the challenge is finished. If the [timeLeft]
+  /// is `null`, the challenge is a group challenge.
+  const ChallengeCardData({
+    required this.title,
+    required this.distance,
+    required this.timeLeft,
+    required this.reward,
+  });
+
+  bool get isFinished => distance == null;
+  bool get isGroupChallenge => timeLeft == null;
```

Comment:
I think you should use extra class attributes instead of mixing this state with distance and time left.

We might decide that group challenges also expire, so a challenge being a group challenge would be independent of time left.

For distance we might temporarily lose location service but still want to display challenges that have been generated previously and that may or may not be complete.

Whether a challenge is completed or not, or is a group challenge or not can be obtained from the database.

Reply:
See my comment below, for the refactoring (done in 77dc241bcca498f09c546c57e4dac1524c3bfe9a). 

> For distance we might temporarily lose location service but still want to display challenges that have been generated previously and that may or may not be complete.
For now, we can just keep the previous GPS position in memory, this is very fine and probably the easiest way to solve many problems when loosing the internet connection. However, if we want to change this in the future, this can be easily done with the refactor (we would, in this case, indeed need some `isFinished` boolean parameter).